### DISCUSSION TITLE
package 修改依赖后生成的路径不变

### DISCUSSION THREAD

**Discussion by TOMO-CAT** [xmake-io/xmake#4712] [Q&A]
### Xmake 版本

xmake v2.8.5

### 操作系统版本和架构

Linux 720ce3a659a2 5.15.90.1-microsoft-standard-WSL2 #1 SMP Fri Jan 27 02:56:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

### 描述问题

定义了一个 lua-protobuf package，然后有其他库 add_requires("lua-protobuf")。修改了 lua-protobuf 依赖的 lua 版本提交到本地的 xmake-repo，然后本地 xmake repo -u 更新仓库。
但是 lua-protobuf 并不会更新，怀疑是 package 配置版本不变（~/.xmake/package后跟着的第二个路径）。
![image](https://github.com/xmake-io/xmake/assets/47598093/1f7e099e-e34c-414d-9154-2a7d3aa71f1c)

### 期待的结果

能不能在识别到 add_deps 版本有变化的时候重新编译，或者提供一个命令更新 package。目前的解法是 rm -rf ~/.xmake/packages/l/lua-protobuf/0db880c0f3e9cbb54d369e1dfbdb2b7f45454876/ 不够优雅。

### 工程配置

```lua
package("lua-protobuf", function()
    add_urls("https://github.com/starwing/lua-protobuf.git")
    add_versions("0db880c0f3e9cbb54d369e1dfbdb2b7f45454876", "fcf56b327cce978397820e06ba8b25810307abc2f8b2d1e72052c919b6262764")

    add_deps("lua v5.4.6")

    on_install(function(package)
        os.cp("protoc.lua", package:installdir("lua_files"))
        io.writefile("xmake.lua", [[
            add_rules("mode.release")
            add_requires("lua v5.4.6")

            target("pb", function()
                set_kind("static")
                add_includedirs(".")
                add_headerfiles("pb.h")
                add_packages("lua")
                add_files("pb.c")
            end)
        ]])
        import("package.tools.xmake").install(package)
    end)
end)
```

### 附加信息和错误日志

![image](https://github.com/xmake-io/xmake/assets/47598093/6d2795c0-2f5a-416f-b23a-482fe900cc6b)

**Comment by samayala22** [comment]
You can try just resetting the current config with `xmake f -c` and then recompile ?

Or maybe force reinstall the package with `xmake require -f -v lua-protobuf` but im not sure if it works with custom package.

**Comment by TOMO-CAT** [comment]
> You can try just resetting the current config with `xmake f -c` and then recompile ?
> 
> Or maybe force reinstall the package with `xmake require -f -v lua-protobuf` but im not sure if it works with custom package.

Thanks, I will try it later and show it's result.

**Comment by waruqi** [comment]
只有修改 add_requires 用户端配置，才会触发包安装，修改包配置，不会触发。。

如果每次修改包都想触发，看下 https://xmake.io/#/zh-cn/guide/build_policies?id=packageinstall_always

**Reply by TOMO-CAT** [reply]
这个配置在没有修改包配置的时候也会重新编译吗？那是不是会导致编译速度问题。

**Reply by waruqi** [reply]
忘记了，你试了就知道了
